addons/hr_holidays/models/hr_leave_stress_day.py

```diff
@@ -34,9 +46,11 @@ def get_stress_days(self, start_date, end_date, resource_calendar_id=None):
                 ('resource_calendar_id', '=', resource_calendar_id.id),
         ])
 
-        for stress_day in stress_days:
-            num_days = (stress_day.end_date - stress_day.start_date).days
-            for d in range(num_days + 1):
-                all_days[str(stress_day.start_date + relativedelta(days=d))] = stress_day.color
+        is_leave_user = self.user_has_groups('hr_holidays.group_hr_holidays_user')
+        if is_leave_user:
+            return stress_days
 
-        return all_days
+        if stress_days.department_ids and department_id:
+            return stress_days.filtered(lambda sd: not sd.department_ids or department_id in sd.department_ids.get_children_department_ids())
+
+        return stress_days
```

Comment:
With lines 50 to 51 removed
```suggestion
        if not is_leave_user and stress_days.department_ids and department_id:
            stress_days = stress_days.filtered(lambda sd: not sd.department_ids or department_id in sd.department_ids.get_children_department_ids())

        return stress_days
```